src/core/thread/mle_router.cpp

```diff
@@ -2522,6 +2543,31 @@ void MleRouter::HandleChildUpdateRequest(const Message &         aMessage,
         ExitNow(error = OT_ERROR_PARSE);
     }
 
+#if OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
+    // CSL
+    if (child->IsCslSynchronized())
+    {
+        CslChannelTlv cslChannel;
+        uint32_t      cslTimeout;
+
+        if (Tlv::FindUint32Tlv(aMessage, Tlv::kCslTimeout, cslTimeout) == OT_ERROR_NONE)
+        {
+            child->SetCslSyncTimeout(cslTimeout);
+        }
+
+        if (Tlv::FindTlv(aMessage, Tlv::kCslChannel, sizeof(cslChannel), cslChannel) == OT_ERROR_NONE)
+        {
+            child->SetCslChannel(static_cast<uint8_t>(cslChannel.GetChannel()));
+        }
+        else
+        {
+            child->SetCslChannel(Get<Mac::Mac>().GetPanChannel());
+        }
+
+        tlvs[tlvslength++] = Tlv::kSourceAddress;
```

Comment:
Do we need to increase the `kMaxResponseTlvs = 10` defined at the top of this method?
(sicne we may now add an additional entry to `tlvs[]` array)

Reply:
I think it shouldn't be here. I have removed it.